components/admin_console/plugin_management/plugin_management.jsx

```diff
@@ -655,6 +684,8 @@ export default class PluginManagement extends AdminSettings {
 
         const uploadBtnClass = enableUploads ? 'btn btn-primary' : 'btn';
 
+        const overwritePluginModal = this.state.showOverwritePluginModal && this.renderOverwritePluginModal();
```

Comment:
This isn't strictly feedback for this review, but I've heard of a modal controller that would help us move away from the components having to own the rendering of the modal. Perhaps other reviewers might have context into this?

Actually, and again not feedback for this review, but it would be really quite elegant to be able to `await` above on a `confirmWithModal` function that spawned a modal and resolved/rejected based on the user's feedback, leaving all the handling of the interaction within the method in question.

Reply:
This seems very reasonable to me, particularly after I struggled to understand where exactly `ConfirmModal` should be invoked from and how. A modal async function for it would make so much sense...